### PR TITLE
Dev/dr/dev server host version

### PR DESCRIPTION
**Pull request by dr1rrb**
linked to https://github.com/unoplatform/uno.ui.runtimetests.engine/issues/102

## Feature
* Force the dev-server to use the desired dotnet runtime
* Add ability to configure hot-reload also when app is built in release

## What is the current behavior?
* The dev-server in run using the latest runtime available, breaking assembly loading and compatibility checks
* Dev server is not configured in release preventing usage of hot-reload

## What is the new behavior?
🙃

### <samp>🤖 Generated by Copilot at bc1ed19</samp>

This pull request adds support for hot reload in release mode with apps built on the CI, and updates the remote control host and client projects to use the latest features of .NET 7.0 and 8.0. It also adds a new property to expose the remote control processors, and cleans up some unused and redundant code. The main files affected are `RemoteControlGenerator.cs`, `Uno.UI.DevServer.targets`, `global-net*.json`, `ClientHotReloadProcessor.cs`, and `RemoteControlClient.cs`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable) ==> In runtime tests project!
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/RemoteControl/RemoteControlGenerator.cs [hunk 1]
```diff
@@ -85,6 +88,29 @@ public void Execute(GeneratorExecutionContext context)
...
+				}
+				if (isServerProcessorsConfigEnabled)
+				{
+					BuildServerProcessorsPaths(context, sb);
+				}
+
+				context.AddSource("RemoteControl", sb.ToString());
+			}
+
+			bool Is(string propertyName)
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/RemoteControl/RemoteControlGenerator.cs:112]: Let's not be minimalistic :)

```suggestion
			bool GetMSBuildPropertyValue(string propertyName)
```

File: src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj [hunk 2]
```diff
@@ -35,6 +35,20 @@
...
+		</Content>
+		<Content Update="Config\global-net7.0.json" Condition="'$(TargetFramework)' == 'net7.0'">
+			<CopyToOutputDirectory>Always</CopyToOutputDirectory>
+			<TargetPath>global.json</TargetPath>
+		</Content>
+		<Content Update="Config\global-net8.0.json" Condition="'$(TargetFramework)' == 'net8.0'">
+			<CopyToOutputDirectory>Always</CopyToOutputDirectory>
+			<TargetPath>global.json</TargetPath>
+		</Content>
+	</ItemGroup>
```
Comment (jeromelaban) [inline src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj:50]: These files are part of https://github.com/unoplatform/uno/blob/master/build/nuget/Uno.WinUI.RemoteControl.nuspec and may need to be modified there as well.

File: src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs [hunk 3]
```diff
@@ -14,357 +14,364 @@
...
+	/// <param name="ct"></param>
+	/// <returns></returns>
+	public Task<bool> WaitForWorkspaceLoaded(CancellationToken ct)
+		=> _hotReloadWorkloadSpaceLoaded.Task;
+
+	/// <summary>
+	/// Determines if the server's hot reload workspace has been loaded
+	/// </summary>
+	internal Task HotReloadWorkspaceLoaded
+		=> _hotReloadWorkloadSpaceLoaded.Task;
```
Comment (jeromelaban) [inline src/Uno.UI.RemoteControl/HotReload/ClientHotReloadProcessor.MetadataUpdate.cs:60]: Let's remove that one and replace call sites.

**Review by jeromelaban — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/RemoteControl/RemoteControlGenerator.cs (hunk 1 above)
Comment (dr1rrb) [reply src/SourceGenerators/Uno.UI.SourceGenerators/RemoteControl/RemoteControlGenerator.cs:112]: It also checks if set to `true` so it should not be `GetXXX` but `IsXXX`  :p

File: src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UI.RemoteControl.Host/Uno.UI.RemoteControl.Host.csproj:50]: The `RemoteControl.nuspec` is for the deprecated package, but I validated it's correct in the DevServer.nupsec : https://github.com/unoplatform/uno/blob/master/build/nuget/Uno.WinUI.DevServer.nuspec#L169 (we are including *.*)

**Review by ajpinedam — APPROVED** [review]

**Pull request merged**